src/blueapi/service/facade.py

```diff
@@ -0,0 +1,85 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+from blueapi.service.model import DeviceModel, PlanModel, WorkerTask
+from blueapi.worker.event import WorkerState
+from blueapi.worker.task import RunPlan
+from blueapi.worker.worker import TrackableTask
+
+
+class BlueskyHandler(ABC):
+    """Interface between web application and underlying Bluesky context and worker"""
+
+    @property
+    @abstractmethod
+    def plans(self) -> List[PlanModel]:
+        """
+        All available plans in the BlueskyContext
+        """
+
+    @abstractmethod
+    def get_plan(self, name: str) -> PlanModel:
+        """
+        Retrieve plan by name from the BlueskyContext
+        """
+
+    @property
+    @abstractmethod
+    def devices(self) -> List[DeviceModel]:
+        """
+        All available devices in the BlueskyContext
+        """
+
+    @abstractmethod
+    def get_device(self, name: str) -> DeviceModel:
+        """
+        Retrieve device by name from the BlueskyContext
+        """
+
+    @abstractmethod
+    def submit_task(self, task: RunPlan) -> str:
+        """
+        Submit a task to be run on begin_task
+        """
+
+    @abstractmethod
+    def delete_task(self, task_id: str) -> str:
```

Comment:
In the worker this is called "clear" rather than "delete" (delete comes from REST api).

Is "clear" better?

Reply:
This only applies to pending tasks. To cancel the active task, `cancel_active_task` is used. So perhaps `clear_pending_task` to remove ambiguity?